spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/TraceIdExtractorCombination.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging;
+
+/**
+ * Enum values represent the order and combination of trace ID extractors that can be
```

Comment:
This enum is used in the LoggingWebMvcConfigurer, and if it's moved to the auto-config/starter project then this project would depend on its starter, which I'm trying to avoid. 

Renaming changes and javadoc changes made.

Reply:
Correct. The `LoggingWebMvcConfigurer` must just don't do that logic. It is really an auto-config responsibility to do configuration by convention. The `LoggingWebMvcConfigurer` should just rely on the injection `Extractor` bean or creating the default one otherwise. No any conventions by properties in the raw module. Just follow the classical Inversion of Control and that's all!